cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/commands/DeployClientConfigCommandRetriever.java

```diff
@@ -0,0 +1,109 @@
+     */
+    public BigDecimal getCommandId(ClustersResourceApi api, Stack stack)
+            throws CloudbreakException, ApiException {
+        ApiResponse<ApiCommandList> commandListResponse =
+                api.listActiveCommandsWithHttpInfo(stack.getName(), null);
+        BigDecimal foundCommandId = getCommandIdFromActiveCommands(commandListResponse);
+        if (foundCommandId == null) {
+            LOGGER.debug("The last deploy client config command could not be found  "
+                    + "by listing active commands. Trying {} call", activeCommandTableResource.getUriPath());
+            List<CommandResource> commandsFromRunningCommandsTable = activeCommandTableResource.getCommands(
+                    api, commandListResponse.getHeaders());
+            LOGGER.debug("Processing activeCommandTable were successful.");
+            foundCommandId = getLatestDeployClientConfigCommandId(
+                    commandsFromRunningCommandsTable, activeCommandTableResource.getUriPath());
+        }
+        if (foundCommandId == null) {
+            LOGGER.debug("The last deploy client config command could not be found  "
+                    + "by listing active commandTable. Trying {} call", recentCommandTableResource.getUriPath());
+            List<CommandResource> commandsFromRecentCommandsTable = recentCommandTableResource.getCommands(
+                    api, commandListResponse.getHeaders());
+            foundCommandId = getLatestDeployClientConfigCommandId(
+                    commandsFromRecentCommandsTable, recentCommandTableResource.getUriPath());
+        }
+        return foundCommandId;
+    }
+
+    @VisibleForTesting
+    BigDecimal getCommandIdFromActiveCommands(ApiResponse<ApiCommandList> response) {
+        if (response.getStatusCode() >= ERROR_CODES_FROM) {
+            return null;
+        }
+        ApiCommandList commandList = response.getData();
+        List<CommandResource> commands = commandList.getItems().stream()
+                .filter(c -> c.getParent() == null)
+                .map(c -> {
+                    CommandResource command = new CommandResource();
+                    command.setStart(new DateTime(c.getStartTime())
+                            .toDate().toInstant().toEpochMilli());
+                    command.setName(c.getName());
+                    command.setId(c.getId().longValue());
+                    command.setSuccess(c.getSuccess());
+                    return command;
+                }).collect(Collectors.toList());
+        return getLatestDeployClientConfigCommandId(commands, "listActiveCommands");
+    }
+
+    private BigDecimal getLatestDeployClientConfigCommandId(
+            List<CommandResource> commandList, String path) {
+        return commandList.stream()
+                .filter(c -> DEPLOY_CLIENT_CONFIG_COMMAND_NAME.equals(c.getName()))
+                .filter(c -> c.getSuccess() != null && c.getSuccess())
```

Comment:
I think you had null check for start time in place in the older version. Is it ok without the null check?